app/assets/v2/js/pages/profile-trust.js

```diff
@@ -2246,4 +2246,174 @@ $(document).ready(function() {
+      const did = '<a href="https://www.w3.org/TR/did-core/" target="_blank">DID</a>';
+
+      return `
+        <p>This is a ${verifiableCredential} that contains your trust bonus signed by the ${did} <strong>${issuer}</strong>.</p>
+        <p>Your trust bonus score is <strong>${value}</strong> for the Ethereum address <strong>${address}</strong>.</p>
+        <textarea style="width: 100%;margin: 0 0 1rem 0;padding: 1rem;" rows="4" class="text-monospace" readonly>${formatted}</textarea>
+        <div style="display: flex;flex-direction: column;">
+          <div style="display: flex;">
+            <button ${style} type="button" class="btn btn-info" id="vc-copy">Copy</button>
+            <a ${style} href="${verifier}" target="_blank" class="btn btn-success">Verify</a>
+            <a ${style} href="${href}" class="btn btn-link" download="gitcoin-popp-vc.json"">Download</a>
+          </div>
+          <button ${style} type="button" class="btn btn-primary" id="vc-3box">Save to 3Box</button>
+        </div>
+      `;
+    };
+
+    const messages = {
+      auth: [
+        'Authenticate with 3Box',
+        'Authenticating with 3Box',
+        'Authenticated with 3Box'
+      ],
+      storage: [
+        `Open '${_3BOX_SPACE}' storage space on 3Box`,
+        `Opening '${_3BOX_SPACE}' storage space on 3Box`,
+        `Opened '${_3BOX_SPACE}' storage space on 3Box`
+      ],
+      variable: [
+        `Save passport credential as '${_3BOX_FIELD}'`,
+        `Saving passport credential as '${_3BOX_FIELD}'`,
+        `Saved passport credential as '${_3BOX_FIELD}'`
+      ]
+    };
+
+    const getMessage = (source, state) => {
+      if (state === true) {
+        return '<s>' + source[2] + '</s> ✓';
+      } else if (state === false) {
+        return '<strong>' + source[1] + '</strong>';
+      }
+      return '<em>' + source[0] + '</em>';
+    };
+
+    const get3BoxContent = (auth, storage, variable) => {
+      let list = '<ul>';
+
+      list += '<li>';
+      list += getMessage(messages.auth, auth);
+      list += '</li>';
+
+      list += '<li>';
+      list += getMessage(messages.storage, storage);
+      list += '</li>';
+
+      list += '<li>';
+      list += getMessage(messages.variable, variable);
+      list += '</li>';
+
+      list += '</ul>';
+
+      return `
+        <div>
+          <p><strong>3Box integration:</strong> you can save a POPP credential to 3Box.</p>
+          ${list}
+        </div>
+      `;
+    };
+
+    const vcPassport = () => {
+      if (document.web3network != 'rinkeby' && document.web3network != 'mainnet') {
+        _alert('Please connect your web3 wallet to mainnet + unlock it', 'danger', 1000);
+        return;
+      }
+
+      const accounts = web3.eth.getAccounts();
+
+      $.when(accounts).then((result) => {
+        ethAddress = result[0];
+        ethProvider = web3.currentProvider;
```

Comment:
there are already some methods used on wallet.js that could be used here instead. You will find some global variables like selectedAccount, provider and others generated when someone connect to the site using web3modal

Reply:
Resolved in https://github.com/gitcoinco/web/pull/9032/commits/e32dea61a3a42ebb5281a21695b4eb15cfa6450a.